bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrDetailsCollector.kt

```diff
@@ -0,0 +1,73 @@
+package com.bugsnag.android
+
+import android.app.ActivityManager
+import android.app.ActivityManager.ProcessErrorStateInfo
+import android.content.Context
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Process
+import android.support.annotation.VisibleForTesting
+import java.util.concurrent.atomic.AtomicInteger
+
+internal class AnrDetailsCollector {
+
+    companion object {
+        private const val INFO_POLL_THRESHOLD_MS: Long = 1000
```

Comment:
Did you try this at a shorter duration? 100ms should be acceptable for the short period of time this poll needs to run.